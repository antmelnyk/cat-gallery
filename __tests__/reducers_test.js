import appReducer, { initialState } from '../src/store/reducer'

describe('appReducer', () => {

  it('has to return the initial state', () => {
    expect(appReducer(undefined, { type: 'Unrecognized' })).toMatchSnapshot()
  })

  it('should handle SET_SELECTED_CATEGORY', () => {
    expect(
      appReducer(initialState,
        {
          type: 'SET_SELECTED_CATEGORY',
          categoryId: 0
        })
    ).toMatchSnapshot()
  })

  it('should handle FETCH_CATEGORIES_SUCCESS', () => {
    expect(
      appReducer(initialState,
        {
          type: 'FETCH_CATEGORIES_SUCCESS',
          categories: [{
            id: 0,
            name: 'Test'
          }],
          pagesLoaded: 0
        })
    ).toMatchSnapshot()
  })

  it('should handle FETCH_IMAGES_SUCCESS', () => {
    expect(
      appReducer(initialState,
        {
          type: 'FETCH_IMAGES_SUCCESS',
          images: [{
            id: 0,
            url: 'url'
          }],
          pagesLoaded: 0,
          imagesFetching: false
        })
    ).toMatchSnapshot()
  })

  it('should handle FETCH_REQUEST_FAILURE', () => {
    expect(
      appReducer(initialState,
        {
          type: 'FETCH_REQUEST_FAILURE',
          error: 'Error'
        })
    ).toMatchSnapshot()
  })

})