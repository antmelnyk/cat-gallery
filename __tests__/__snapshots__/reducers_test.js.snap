// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appReducer has to return the initial state 1`] = `
Object {
  "categories": Array [],
  "error": "",
  "images": Array [],
  "imagesFetching": false,
  "pagesLoaded": 0,
  "selectedCategory": null,
}
`;

exports[`appReducer should handle FETCH_CATEGORIES_SUCCESS 1`] = `
Object {
  "categories": Array [
    Object {
      "id": 0,
      "name": "Test",
    },
  ],
  "error": "",
  "images": Array [],
  "imagesFetching": false,
  "pagesLoaded": 0,
  "selectedCategory": null,
}
`;

exports[`appReducer should handle FETCH_IMAGES_SUCCESS 1`] = `
Object {
  "categories": Array [],
  "error": "",
  "images": Array [
    Object {
      "id": 0,
      "url": "url",
    },
  ],
  "imagesFetching": false,
  "pagesLoaded": 1,
  "selectedCategory": null,
}
`;

exports[`appReducer should handle FETCH_REQUEST_FAILURE 1`] = `
Object {
  "categories": Array [],
  "error": "Error",
  "images": Array [],
  "imagesFetching": false,
  "pagesLoaded": 0,
  "selectedCategory": null,
}
`;

exports[`appReducer should handle SET_SELECTED_CATEGORY 1`] = `
Object {
  "categories": Array [],
  "error": "",
  "images": Array [],
  "imagesFetching": false,
  "pagesLoaded": 0,
  "selectedCategory": 0,
}
`;
